src/services/validation.status.service.ts

```diff
@@ -15,5 +16,12 @@ export const getValidationStatus = async (session: Session, transactionId: strin
       throw new Error(`Error retrieving validation status: ${JSON.stringify(errorResponse)}`);
     }
     const successfulResponse = response as Resource<ValidationStatusResponse>;
-    return successfulResponse.resource as ValidationStatusResponse;
+    return logValidationErrors(successfulResponse.resource as ValidationStatusResponse);
+}
+
+const logValidationErrors = (validationStatus: ValidationStatusResponse): ValidationStatusResponse => {
+    if (!validationStatus.isValid) {
+      logger.error("Validation errors: " + JSON.stringify(validationStatus.errors.slice(0, 10)));
```

Comment:
Should this be debug level logging? Could generate significant amount of logging as we call this on a number of web screens.

Reply:
But we only log if we have errors and then go on to fail. I've limited the actual errors to 10 but we'd want to investigate any errors returned by the API as the Web should be checking and not passing invalid data to the API.

Reply:
We typically don't use info or warn level and debug can be turned off / ignored to better to deal with these API invalid returns as errors.

Reply:
> But we only log if we have errors and then go on to fail. I've limited the actual errors to 10 but we'd want to investigate any errors returned by the API as the Web should be checking and not passing invalid data to the API.

Yes good point, I forgot we have removed the API call for most web pages. We still seem to call it here which will create some noise due to missing fields as address comes before some mandatory fields in the web journey. We got around this by filtering after the API call is made to avoid showing these errors on the front end however I believe this logging will occur before the filtering. 
https://github.com/companieshouse/officer-filing-web/blob/ba6b5dc8a225d96ebfaccd0551e9da0740341ec5/src/controllers/shared.controllers/director.correspondence.address.manual.controller.ts#L125

Reply:
we've already validated the address in the Web app by calling validateManualAddress so I think it would be good to log any differences with the API validation address those. So all good I think.

Reply:
You will get errors every time that page is posted, in the AP01 flow it will list things such as protected details being missing for instance as that page comes after the address flow.

Reply:
I've made the logging of errors optional (default true) and suppressed it for this call.